lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -49,18 +47,26 @@ export class AzureArc extends BaseManagedIdentitySource {
         nodeStorage: NodeStorage,
         networkClient: INetworkModule,
         cryptoProvider: CryptoProvider,
-        identityEndpoint: string
+        identityEndpoint: string,
+        systemAssigned: boolean
```

Comment:
nit: As stated before, it's better to pass in the full configuration about SAMI / UAMI as it might help with logging etc. There's no need for higher level class to derive a flag and inject it here.